python/model_hosting_container_standards/common/transforms/utils.py

```diff
@@ -45,3 +73,63 @@ def set_value(obj: dict, path: str, value: Any) -> dict:
     # Assign directly (since parent is a dict)
     parent[child] = value
     return obj
+
+
+def _set_value_with_parent_creation(
+    obj: dict,
+    parent_parts: list[str],
+    child,
+    value,
+    max_create_depth: int = DEFAULT_MAX_DEPTH_TO_CREATE,
+    _full_depth: Optional[int] = None,
+):
+    """Set a value in a nested dict, creating parent structures as needed.
+
+    Args:
+        obj: The dictionary to modify
+        parent_parts: List of parent keys to traverse/create
+        child: The final child key to set
+        value: The value to set at the child key
+        max_create_depth: Maximum nesting depth of the full path when creating parents (None = unlimited). Defaults to DEFAULT_MAX_DEPTH_TO_CREATE.
+        _full_depth: Internal parameter tracking the total depth of the full path
+
+    Returns:
+        The modified obj dictionary
+
+    Raises:
+        KeyError: If max_create_depth is exceeded when creating parents
+    """
+    if len(parent_parts) == 0:
+        obj[child] = value
+        return obj
+
+    parent_expr = ".".join(parent_parts)
+    parent = jmespath.search(parent_expr, obj)
```

Comment:
`jmespath.search` is O(K) where K is number of keys (inc. nested) in obj
With recursion this becomes O(KD) where D is the max_create_depth